pygmt/helpers/decorators.py

```diff
@@ -574,6 +565,14 @@ def new_module(*args, **kwargs):
                         f"Parameters in short-form ({short_param}) and "
                         f"long-form ({long_alias}) can't coexist."
                     )
+                if (long_alias in kwargs and long_alias in ["xshift", "yshift"]) or (
+                    short_param in kwargs and short_param in ["X", "Y"]
+                ):
+                    raise GMTInvalidInput(
+                        f"Parameters ({short_param}) and "
+                        f"({long_alias}) are not supported anymore."
+                        f" Please use shift_origin() instead!"
+                    )
```

Comment:
Using an error here will immediately break people's code if they're using `xshift` or `yshift`. Better to go through a deprecation cycle that raises a `FutureWarning` instead.

One idea is to use the [`@deprecate_parameter`](https://github.com/GenericMappingTools/pygmt/blob/d80b986a3f5634e2e68d98d569405dfa3eba1475/pygmt/helpers/decorators.py#L732-L808) function and do something like `@deprecate_parameter(oldname="xshift", newname=None, deprecate_version="v0.8.0", remove_version="0.10.0")`. This will require some modifications to the decorator to handle `newname=None` though.

Reply:
> One idea is to use the [`@deprecate_parameter`](https://github.com/GenericMappingTools/pygmt/blob/d80b986a3f5634e2e68d98d569405dfa3eba1475/pygmt/helpers/decorators.py#L732-L808) function and do something like `@deprecate_parameter(oldname="xshift", newname=None, deprecate_version="v0.8.0", remove_version="0.10.0")`. This will require some modifications to the decorator to handle `newname=None` though.

We don't have to use `@deprecate_parameter` for this, which requires adding `@deprecate_parameter` decorators in many Python files.

Reply:
I modified the passage @seisman. I understand the concerns @weiji14 has regarding raising an error here. However, we did agree to totally drop this option (not only deprecate it), didn't we?

Reply:
> was there a discussion to drop xshift/yshift in a backward incompatible way?

No, we definitely need **backward compatibility**.

The final goal is, for the next 2 or more minor versions, users can still use parameters `xshift`, `yshift`, `X`, `Y`, and they will see a *FutureWarning* like:
```
Parameters xshift, yshift, X and Y are deprecated and will be no longer 
supported in v0.x.x. Please user Figure.shift_origin() instead.
```

> One idea is to use the [`@deprecate_parameter`](https://github.com/GenericMappingTools/pygmt/blob/d80b986a3f5634e2e68d98d569405dfa3eba1475/pygmt/helpers/decorators.py#L732-L808) function and do something like `@deprecate_parameter(oldname="xshift", newname=None, deprecate_version="v0.8.0", remove_version="0.10.0")`. This will require some modifications to the decorator to handle `newname=None` though.

I believe this will work but it's not ideal. It's easy to modify the `deprecate_parameter` decorator to support `newname=None`, but the warning message would be something like:
```
The 'xxxx' parameter has been deprecated since vx.x.x and will be removed in vx.x.x.
```
The error message can't recommend using `Figure.shift_origin()` unless we deal with `xshift` and `yshfit` in a special way.

It's also more confusing when users use the short parameters `X` or `Y`. When they use `X`, they are recommended to use `xshift`. After they modify `X` to `xshift`, then they see `xshfit` is deprecated.

Reply:
I've opened PR #2135, which shows how we can deprecate `timestamp` (`U`) in a backward-compatible and recommend `Figure.timestamp()` (not implemented yet) instead. The same idea can be applied to xshift and yshift in this PR.

Reply:
I applied your implemetations for this PR @seisman.